UM-USElab/gradecraft-development
# 1534 assignment copy attachments

### Status
IN DEVELOPMENT

### Description
This PR adds the ability to copy assignment files on S3 when copying assignments on Gradecraft.

### Gem dependencies
N/A

### Migrations
YES

### Steps to Test or Reproduce
Add assignment file(s) to any assignment and create a copy. Ensure that the assignment files for the original are copied over to a location relative to the copied course.

For example:

`jon_yu/uploads/GC101-1/assignments/accepts_all_types_of_-151/assignment_files/1497375959_not_a_real_assignment_upload.txt`

should be copied to

`jon_yu/uploads/GC101-1/assignments/copy_of_accepts_all_t-496/assignment_files/1497375959_not_a_real_assignment_upload.txt`

Observe the differing assignment names where the term "copy_of" has been prepended to the duplicate assignment.

### Impacted Areas in Application
Assignment copying

======================
Closes #1534
